### ISSUE TITLE
Addresses' lat/lon coordinates

### ISSUE THREAD

**Issue by giacecco** [OpenAddressesUK/forum#3]
It is already among Open Addresses UK's objectives to make the dataset easily extensible by third parties, but should lat/lon coordinates for the addresses be part of the default dataset? See also Angharad Stone's request at https://twitter.com/AngharadStone/status/549165105452421120 .

**Comment by TimSC** [comment]
This seems like an obvious feature! Even as an optional data field.

If people are going to add a location on a map, I might suggest OS OpenData StreetView because it will minimize restrictions on further use. Maps like OSM or fosm are share-alike.

**Comment by peterkwells** [comment]
Thanks TimSC, OS OpenData Streetview might be a good option. The resulting dataset needs to be releasable under, at most, an attribution licence.

**Comment by ChrisMewse** [comment]
Agree that lat/lon should be provided, and without sounding too needy, as it's a UK dataset British National Grid Easting and Northings.

As discussed with Peter (IRL) I'd like to experiment with the option of using aerial imagery as this could be completely royalty/attribute free and give users the option to position their property more accurately,

**Comment by TimSC** [comment]
@ChrisMewse If you require national grid references, use the OSTN02 transform to convert from lat/lon - it's now available in several programming languages (I ported a perl version into PHP and Python). Lat/lon is widely used in software and navigation devices (usually WGS84).

https://github.com/TimSC/ostn02python
https://github.com/TimSC/ostn02php

**Comment by jeremy45** [comment]
Indeed, lat/lon for the actual addresses would be a very useful and valuable dataset.

**Comment by peterkwells** [comment]
Just a quick update on this one as we've had a few emails/tweets as well. Current position below, as ever things can change.....

---
1. At this point in time we have chosen to focus on addresses and the building blocks that form addresses (postcodes, towns, streets, &c) rather than more detailed lat/lon data as we think it's the best route to the sustainable business model we need (i.e. funding).
2. Our open data source for postcodes is the Office for National Statistics’ “Postcode Directory” (the latest edition is at https://geoportal.statistics.gov.uk/geoportal/catalog/search/resource/details.page?uuid=%7BDA90AA42-20E0-4A0C-9AB7-DC4F518385DB%7D). We preserve the postcode centroids’ coordinate information, as its useful, reliable and consistent with our licensing. The only thing we do is convert it from northing/easting to lat/lon.
3. Clearly postcode centroids coordinates satisfy many use cases but not all....
4. We  all might want to satisfy those needs in the future but for the time being we are choosing to stay focussed on the first task: building the main address list and the building blocks that form it.
5. The platform has been deliberately designed to be extensible so that we can add detailed coordinates for individual addresses in the future. That could be done by bringing in other large sources; opening up APIs for others to add; or by extending whatever crowdsourcing services we have at the time.

---

**Comment by jeremy45** [comment]
Peter,
Thanks for the update. Indeed postcode locations are useful. However I personally don't think accurate enough for my purposes. 
Regards. 
Jeremy.

**Comment by samuelleach** [comment]
Just to say that I think that having the latest known central lon lat in WGS84 coords available in the downloadable CSV would greatly enhance the use of the data for 'proximity' use cases, as well as exploratory analyses.

**Comment by TimSC** [comment]
There is nothing stopping a separate project to add lat/lon to OpenAddressesUK data. Alternatively, the lat, lon positions could be in openstreetmap.org or fosm.org with a reference to an OpenAddressesUK record.

**Comment by peterkwells** [comment]
@TimSC is absolutely right. Our URI could be used as a link although note that we may tweak the current ones soon. After that we'll lock it in: https://github.com/OpenAddressesUK/roadmap/issues/6

Obviously Ordnance Survey have also announced that the UPRN (unique property reference number) is becoming open data but there may still be some issues with allocating new ones or discovering the UPRNs for existing data. Maybe something to work out in a Discovery phase.....
